### DISCUSSION TITLE
Feature Request: ROS2 launch file for Foxglove Studio

### DISCUSSION THREAD

**Discussion by tonynajjar** [foxglove/community#982] [General]
It would be quite useful to launch Foxglove Studio automatically with a ROS2 stack. For this we would need a launch file from Foxglove Studio. Ideally we could specify the data source and the layout from the launch configuration

**Comment by defunctzombie** [comment]
Look into https://docs.foxglove.dev/docs/visualization/shareable-links/ which has some notes about how to launch the desktop app (and web app) with a specific data source and layout.

Creating the launch file is outside the scope of what we would deliver since it is specific to your personal workflow needs - different folks want to launch things in different ways. Some users don't close Foxglove and run it separately from their ROS stack launches.

**Comment by defunctzombie** [comment]
Related community slack thread: https://foxglove.slack.com/archives/C01T10L5QJX/p1711464396025809

**Comment by defunctzombie** [comment]
@tonynajjar For your workflow are you hoping to open to a live local connection (via websocket) or a local file?

**Reply by tonynajjar** [reply]
For my workflow a live connection

**Comment by defunctzombie** [comment]
Internal issue: https://linear.app/foxglove/issue/FG-7266/opening-a-sharable-link-url-does-not-work-on-linux

**Reply by tonynajjar** [reply]
what's the status of this ticket on your side

**Reply by defunctzombie** [reply]
Ticket was done 1 year ago. Are you still having problems opening share links?

**Reply by tonynajjar** [reply]
Didn't try, just wanted to clean up. I'll close then
